core/iwasm/interpreter/wasm_interp_classic.c

```diff
@@ -871,17 +874,35 @@ wasm_interp_call_func_import(WASMModuleInstance *module_inst,
        this function */
     prev_frame->ip = NULL;
 
-    /* replace exec_env's module_inst with sub_module_inst so we can
-       call it */
+    /* Switch exec_env but keep using the same one by replacing necessary
+     * variables */
+    sub_module_exec_env = wasm_runtime_get_exec_env_singleton(
+        (WASMModuleInstanceCommon *)sub_module_inst);
+    if (!sub_module_exec_env) {
+        wasm_set_exception(module_inst, "create singleton exec_env failed");
+        return;
```

Comment:
Can we put L873-L875 `prev_frame->ip = NULL;`  after L 885, or here restore the `prev_frame->ip` before return? So that it isn't changed when exception is thrown, if exception handling is enabled, `prev_frame->ip` might be used after handling the exception.